c_src/image/image.c

```diff
@@ -58,10 +58,6 @@ void image_free(void* v_ctx, image_t* p_image)
     tommy_hashlin_remove_existing(&images, &p_image->node);
     image_ops_delete(v_ctx, p_image->image_id);
 
-    if (p_image->p_pixels) {
-      free(p_image->p_pixels);
-    }
```

Comment:
Are we sure we want to remove this? It'll cause a memory leak, no?

Reply:
Yes, we want to remove this. `p_image` is malloc'd here: https://github.com/ScenicFramework/scenic_driver_local/blob/main/c_src/image/image.c#L208 and `p_image->p_pixels` is a pointer inside of that memory (see line 227 of that same function). With this code doing a free, it causes free to blow up because that pointer was never malloc'd

Reply:
I'm looking around `226` at:

```C

    // initialize the pixel pointer
    p_image->p_pixels = ((void*)p_image) + struct_size + id_size;
    ```
    
It seems like it points at allocated memory somewhere, right--even though freeing it fails since it's been moved ahead of the bookkeeping information malloc is probably using internally. Am I missing something?

(And thanks for double-checking my work!)

Reply:
The memory allocated for `p_image` includes the memory that will be used by what is pointed to by `p_image->p_pixels`:
```
    int struct_size = ALIGN_UP(sizeof(image_t), 8);
    // the +1 is so the id is null terminated
    int id_size = ALIGN_UP(id_length + 1, 8);
    int pixel_size = width * height * 4;
    int alloc_size = struct_size + id_size + pixel_size;

    p_image = malloc(alloc_size);
```
The only thing that needs to be `free()`d is the `p_image` pointer. It is a bug to also try to `free()` the `p_image->p_pixels` pointer. This  code needs to be removed.

Reply:
Gotcha; I'm convinced now after reading the code more closely. Thank you!